shared/common-adapters/avatar.render.native.js

```diff
@@ -112,44 +95,42 @@ class AvatarRender extends React.PureComponent<Props, State> {
   render() {
     const {size} = this.props
     const borderRadius = this.props.isTeam ? sizeToTeamBorderRadius[String(size)] : size / 2
-    const containerStyle = collapseStyles([styles[`box:${size}`], this.props.style])
+    const containerStyle = Styles.collapseStyles([styles[`box:${size}`], this.props.style])
 
     return (
-      <ClickableBox onClick={this.props.onClick} feedback={false} style={containerStyle}>
-        <Box style={containerStyle}>
-          {!this.props.skipBackground && (!this.props.skipBackgroundAfterLoaded || !this.state.loaded) && (
-            <Background
-              loaded={this.state.loaded}
-              loadingColor={this.props.loadingColor}
-              borderRadius={borderRadius}
-            />
-          )}
-          {!!this.props.url && (
-            <UserImage
-              opacity={this.props.opacity}
-              onLoadEnd={this._onLoadOrError}
-              size={size}
-              url={this.props.url}
-              borderRadius={borderRadius}
-            />
-          )}
-          {(!!this.props.borderColor || this.props.isTeam) && (
-            <Border
-              borderColor={this.props.borderColor || globalColors.black_10}
-              borderRadius={borderRadius}
-            />
-          )}
+      <Kb.ClickableBox onClick={this.props.onClick} feedback={false} style={containerStyle}>
+        <Kb.Box style={containerStyle}>
+          {!this.props.skipBackground &&
+            (!this.props.skipBackgroundAfterLoaded || !this.state.loaded) &&
+            background({
```

Comment:
Capitalize and treat like components

Reply:
These are not components on purpose

Reply:
Ooooh ok TIL the difference. Why would we ever use jsx over this for functional components? I guess for hooks, now. Well anyway sounds good.

Reply:
this isn't really a sub component where some of its props might change. so making it a real component like it used to do just creates more overhead. if you look at desktop its all just one function. this means if the parent renders this does too but thats what happens anyways practically and its a lot faster to just call it and not have this also mount